Patch-Source: https://gitlab.gnome.org/GNOME/epiphany/-/commit/84474398f6e59266b73170838219aa896729ce93.patch

From 84474398f6e59266b73170838219aa896729ce93 Mon Sep 17 00:00:00 2001
From: Patrick Griffis <pgriffis@igalia.com>
Date: Thu, 23 Dec 2021 14:51:50 -0600
Subject: [PATCH] Port to libportal 0.5

This was an API breaking release that requires minor changes.
---
 lib/ephy-flatpak-utils.c          | 2 +-
 meson.build                       | 2 +-
 org.gnome.Epiphany.Canary.json.in | 7 +++++--
 org.gnome.Epiphany.json           | 7 +++++--
 4 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/lib/ephy-flatpak-utils.c b/lib/ephy-flatpak-utils.c
index 818af95..aef937f 100644
--- a/lib/ephy-flatpak-utils.c
+++ b/lib/ephy-flatpak-utils.c
@@ -29,7 +29,7 @@
 #include <gio/gio.h>
 #include <gio/gunixfdlist.h>
 #if USE_LIBPORTAL
-#include <libportal/portal-gtk3.h>
+#include <libportal-gtk3/portal-gtk3.h>
 #endif
 #include <string.h>
 #include <sys/stat.h>
diff --git a/meson.build b/meson.build
index 74eee8a..40dc740 100644
--- a/meson.build
+++ b/meson.build
@@ -96,7 +96,7 @@ libhandy_dep = dependency('libhandy-1', version: '>= 1.1.0')
 libsecret_dep = dependency('libsecret-1', version: '>= 0.19.0')
 libxml_dep = dependency('libxml-2.0', version: '>= 2.6.12')
 nettle_dep = dependency('nettle', version: nettle_requirement)
-portal_dep = dependency('libportal', version: '>= 0.0.2', required: get_option('libportal'))
+portal_dep = dependency('libportal-gtk3', version: '>= 0.5', required: get_option('libportal'))
 sqlite3_dep = dependency('sqlite3', version: '>= 3.22')
 
 if get_option('soup2').enabled()
diff --git a/org.gnome.Epiphany.Canary.json.in b/org.gnome.Epiphany.Canary.json.in
index 17d4755..f7ffa12 100644
--- a/org.gnome.Epiphany.Canary.json.in
+++ b/org.gnome.Epiphany.Canary.json.in
@@ -106,12 +106,15 @@
             "name" : "libportal",
             "buildsystem" : "meson",
             "config-opts" : [
-                "-Dgtk_doc=false"
+                "-Dbackends=gtk3",
+                "-Dintrospection=false",
+                "-Ddocs=false"
             ],
             "sources" : [
                 {
                     "type" : "git",
-                    "url" : "https://github.com/flatpak/libportal.git"
+                    "url" : "https://github.com/flatpak/libportal.git",
+                    "tag": "0.5"
                 }
             ]
         },
diff --git a/org.gnome.Epiphany.json b/org.gnome.Epiphany.json
index 77993c1..f45c1db 100644
--- a/org.gnome.Epiphany.json
+++ b/org.gnome.Epiphany.json
@@ -103,12 +103,15 @@
             "name" : "libportal",
             "buildsystem" : "meson",
             "config-opts" : [
-                "-Dgtk_doc=false"
+                "-Dbackends=gtk3",
+                "-Dintrospection=false",
+                "-Ddocs=false"
             ],
             "sources" : [
                 {
                     "type" : "git",
-                    "url" : "https://github.com/flatpak/libportal.git"
+                    "url" : "https://github.com/flatpak/libportal.git",
+                    "tag": "0.5"
                 }
             ]
         },
-- 
2.34.1

